### PR TITLE
Fix AudioStreamPlayer iOS 9.0 crash

### PR DESCRIPTION
**Pull request by JagCesar**
We were using API's not available in iOS 9.0, which made our SDK crash.
Got rid of these, and re-implemented similar functionality using
existing API's.

issue #49

### CODE REVIEW

**Comment by JagCesar** [comment]
@Boerworz Can you have a look at this when u got some time? :)

File: Allihoopa/Browse/AudioStreamPlayer.m [hunk 1]
```diff
@@ -175,7 +180,21 @@ -(void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NS
...
+				[self addObservers];
+				// I think we should either inform the user an error occured and the piece couldn't be player
+				// Or try to play the piece again (and maybe get stuck in a loop)
+			} else if ([player status] == AVPlayerStatusReadyToPlay && [self targetPlayingState] == AHATargetPlayingStatePlay) {
+				// Does this mean we should set the state to play?
+				[self setStatus:AHAAudioStreamPlayerStatusPlaying];
+				player.rate = 1;
+			}
+		} else if([keyPath isEqualToString:AHAKeyPath(player, rate)]) {
+
```
Comment (Boerworz) [inline Allihoopa/Browse/AudioStreamPlayer.m]: Perhaps we don't need to observe `rate` on the player if we don't react to the change?

File: Allihoopa/Browse/AudioStreamPlayer.m [hunk 2]
```diff
@@ -175,7 +180,21 @@ -(void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NS
...
-		if ([keyPath isEqualToString:AHAKeyPath(player, currentItem)]) {
+		if ([keyPath isEqualToString:AHAKeyPath(player, status)]) {
+			if (player.status == AVPlayerStatusFailed) {
+				[self removeObservers];
+				self.audioPlayer = [self createPlayer];
+				[self addObservers];
+				// I think we should either inform the user an error occured and the piece couldn't be player
+				// Or try to play the piece again (and maybe get stuck in a loop)
+			} else if ([player status] == AVPlayerStatusReadyToPlay && [self targetPlayingState] == AHATargetPlayingStatePlay) {
+				// Does this mean we should set the state to play?
```
Comment (Boerworz) [inline Allihoopa/Browse/AudioStreamPlayer.m]: Who are you asking? :smiley: Maybe this comment should be removed.

File: Allihoopa/Browse/AudioStreamPlayer.m [hunk 3]
```diff
@@ -175,7 +180,21 @@ -(void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NS
...
 		AVPlayer *player = self.audioPlayer;
 
-		if ([keyPath isEqualToString:AHAKeyPath(player, currentItem)]) {
+		if ([keyPath isEqualToString:AHAKeyPath(player, status)]) {
+			if (player.status == AVPlayerStatusFailed) {
+				[self removeObservers];
+				self.audioPlayer = [self createPlayer];
+				[self addObservers];
+				// I think we should either inform the user an error occured and the piece couldn't be player
+				// Or try to play the piece again (and maybe get stuck in a loop)
```
Comment (Boerworz) [inline Allihoopa/Browse/AudioStreamPlayer.m]: Was this just a note to yourself?

Comment (JagCesar) [reply Allihoopa/Browse/AudioStreamPlayer.m]: To all of us. It's something we should fix, but there's no time to do it now :P

File: Allihoopa/Browse/AudioStreamPlayer.m (hunk 2 above)
Comment (JagCesar) [reply Allihoopa/Browse/AudioStreamPlayer.m]: Yeah, this was a note. We do not want to play when `ReadyToPlay` is called.

File: Allihoopa/Browse/AudioStreamPlayer.m (hunk 1 above)
Comment (JagCesar) [reply Allihoopa/Browse/AudioStreamPlayer.m]: Yes.

**Pull request merged**